modules/internal/ChapelBase.chpl

```diff
@@ -1286,23 +1306,29 @@ module ChapelBase {
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0.0 then
+        halt("Attempt to divide by zero");
```

Comment:
(meanwhile, I'm doing a quick check to see if I can figure out why the / version has a param/non-param overload.  If not, I'll remove it.  If there is a good reason, we may need to do the same here (but I hope not)).